primitives/traits/src/lib.rs

```diff
@@ -72,11 +72,42 @@ pub trait GetCurrentContainerChains {
     fn set_current_container_chains(container_chains: &[ParaId]);
 }
 
+/// How often should a parathread collator propose blocks.
+#[derive(Clone, Debug, Encode, Decode, scale_info::TypeInfo, PartialEq, Eq)]
+pub struct SlotFrequency {
+    /// The parathread will produce at most 1 block every x slots. min=10 means that collators can produce 1 block
+    /// every `x >= 10` slots, but they are not enforced to. However they cannot produce a block after less than 10
+    /// slots, it will be invalid.
+    // TODO: invalid how, slashing collator? Not paying block reward?
```

Comment:
for now only not paying reward

Reply:
Also this struct is ignored, I will add some comment to reflect that.